src/calendar/MonthPicker.ts

```diff
@@ -0,0 +1,299 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v, w } from '@dojo/widget-core/d';
+import uuid from '@dojo/core/uuid';
+import { Keys } from '../common/util';
+import { DNode, TypedTargetEvent } from '@dojo/widget-core/interfaces';
+import Radio from '../radio/Radio';
+import Button from '../button/Button';
+import * as css from './styles/calendar.m.css';
+import * as baseCss from '../common/styles/base.m.css';
+
+// will need to localize messages
+export interface CalendarMessages {
+	chooseMonth: string;
+	chooseYear: string;
+	previousMonth: string;
+	nextMonth: string;
+};
+
+/**
+ * @type MonthPickerProperties
+ *
+ * Properties that can be set on a Calendar component
+ *
+ * @property labelId              Set id to reference label containing current month and year
+ * @property labels               Customize or internationalize accessible helper text
+ * @property month                Currently displayed month, zero-based
+ * @property monthNames           Array of full and abbreviated month names
+ * @property open                 Boolean, sets state of popup
+ * @property renderMonthLabel     Format the displayed current month and year
+ * @property year                 Currently displayed year
+ * @property onRequestOpen        Called when a user action occurs that should trigger the popup opening
+ * @property onRequestClose       Called when a user action occurs that should close the popup
+ * @property onRequestMonthChange Called when a month should change; receives the zero-based month number
+ * @property onRequestYearChange  Called when a year should change; receives the year as an integer
+ */
+export interface MonthPickerProperties extends ThemeableProperties {
+	labelId?: string;
+	labels: CalendarMessages;
+	month: number;
+	monthNames: { short: string; long: string; }[];
+	open?: boolean;
+	renderMonthLabel?(month: number, year: number): string;
+	year: number;
+	onRequestOpen?(): void;
+	onRequestClose?(): void;
+	onRequestMonthChange?(month: number): void;
+	onRequestYearChange?(year: number): void;
+};
+
+export const MonthPickerBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class MonthPicker extends MonthPickerBase<MonthPickerProperties> {
+	private _buttonId = uuid();
+	private _dialogId = uuid();
+	private _labelId = uuid();
+	private _yearSpinnerId = uuid();
+	private _radiosName = uuid();
+
+	private _callTriggerFocus = false;
+	private _callPopupFocus = false;
+
+	// move focus when opening/closing the popup
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		// TODO: find a way to call focus on button widget
```

Comment:
The problem here (and below with the radio buttons) is it involves handling focus and focusability within other widgets. I could either modify all form widgets to be able to be focused and add `tabIndex` to properties or get a working focus manager. I'm inclined towards the latter, but if Calendar needs to wait for a focus manager it'll be here a while :P. I could update the comment to say it needs to be fixed once the focus manager lands and put in a link to the issue